hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -225,9 +223,18 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 	switch nodePool.Spec.Platform.Type {
 	case hyperv1.AWSPlatform:
 		md := machineDeployment(nodePool, hcluster.Spec.InfraID)
-		ami, err := r.ImageProvider.Image(hcluster)
-		if err != nil {
-			return ctrl.Result{}, fmt.Errorf("failed to obtain AMI: %w", err)
+		var ami string
```

Comment:
@enxebre @sjenning should we be using the value from spec if provided and only discovering if no value was given?

Reply:
yes.

Reply:
Actually this lookup needs to be done from the MCS version used for this nodePool, which is the one dictating the ign payload and therefore the nodePool version

Reply:
It makes sense — although this is platform specific, and there's currently no struct type in MCS.Status to express that sort of thing. So maybe something like:

```yaml
status:
  version: ...
  platform:
    type: aws
    aws:
      ami: ...
```

Or something... the info is already embedded in NodePool via NodePoolPlatform, so not sure if it makes sense to embed a NodePoolPlatform in MCS.Status?

Reply:
I also added some TODOs around some possible gaps in the NodePool API here (AWS region, architecture).

I think we should move the status API additions to a followup if that's okay